torch_ttnn/passes/to_tt_pass.py

```diff
@@ -364,20 +375,50 @@ def ReplaceMoreTtManually(gm: torch.fx.GraphModule) -> torch.fx.GraphModule:
                     "memory_config": memory_config,
                 }
                 new_node = g.call_function(ttnn.as_tensor, args, new_kwargs)
+                new_node.meta["val"] = node.meta["val"]
                 node.replace_all_uses_with(
                     new_node,
                     delete_user_cb=lambda node: node != new_node,
                 )
+            if node.target == torch.ops.aten.expand.default:
+                # aten.expand and ttnn.repeat has different meaning for their `shape` argument
+                # aten.expand: the desired output shape, where respective singleton dims are broadcasted
+                # ttnn.repeat: the number of times to repeat a respective singleton dim
+                input_tensor_shape = args[0].meta["val"].size()
+                output_shape = torch.Size(args[1])
+
+                multiplier = np.array(output_shape) // np.array(input_tensor_shape)
+
+                shape_node = g.call_function(ttnn.Shape, (multiplier.tolist(),), {})
+                new_node = g.call_function(ttnn.repeat, (args[0], shape_node), {})
+                new_node.meta["val"] = node.meta["val"]
+                node.replace_all_uses_with(
+                    new_node, delete_user_cb=lambda node: node != new_node
+                )
 
     gm = GraphCleanup(gm)
     return gm
 
 
 class ToTtPass(PassBase):
     def call(self, gm: torch.fx.GraphModule):
+        # TODO(kevinwuTT): transform() removes metadata from Placeholder (input argument) nodes.
+        # Reversing the order will not work either because of the already converted ttnn.ops under
+        # ReplaceMoreTtManually. Might need to consider consolidating these methods.
+
+        # Save Placeholder metadata before calling transform()
+        input_nodes = {
+            node.name: node.meta for node in gm.graph.nodes if node.op == "placeholder"
+        }
+
         # Replace more patterns with torch.fx.Transformer
         gm = ReplaceMoreTt(gm).transform()
 
+        # Restore metadata for Placeholder nodes
```

Comment:
why do we lose metadata?

Reply:
From my investigations so far, it could be an oversight. I found some very recent discussion https://discuss.pytorch.org/t/understanding-torch-fx-traceback-preserve-node-meta/205319 with the same issue but unfortunately no response yet. I could create an issue to track this actually. Could be something we optimize away at a later time if we find a better solution.

Reply:
https://github.com/tenstorrent-metal/pytorch2.0_ttnn/issues/22